widget/hyperlink.go

```diff
@@ -0,0 +1,140 @@
+package widget
+
+import (
+	"image/color"
+	"net/url"
+	"os/exec"
+	"runtime"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/theme"
+)
+
+// Hyperlink widget is a text component with appropriate padding and layout.
+// When clicked, the default web browser should open with a URL
+type Hyperlink struct {
+	textProvider
+	Text      string
+	Url       string
+	Alignment fyne.TextAlign // The alignment of the Text
+	TextStyle fyne.TextStyle // The style of the hyperlink text
+}
+
+// NewHyperlink creates a new layout widget with the set text content
+func NewHyperlink(text string, sUrl string) *Hyperlink {
+	// ignore the error here, we'll continue anyway
+	return NewHyperlinkWithStyle(text, sUrl, fyne.TextAlignLeading, fyne.TextStyle{})
+}
+
+// NewHyperlinkWithStyle creates a new layout widget with the set text content
+func NewHyperlinkWithStyle(text string, sUrl string, alignment fyne.TextAlign, style fyne.TextStyle) *Hyperlink {
+	hl := &Hyperlink{
```

Comment:
so, is Hyperlink 1 word or 2? the short for hl implies 2, but the capitalisation is just 1 - can this be consistent? I think h := would be fine.

Reply:
1 word, but 2 halves (in my opinion).  I can change them to `h`.

Reply:
An advantage to using a variable like `hl` is I can easily find all instances of `hl` in a file, but just finding instances of `h` is obviously nearly impossible.  I could find+replace one direction, but not the other.

Reply:
That is a good point. I suppose it's not really part of the API contract (though it will form part of the documentation). I don't know if go style guide wants single letter of if its just convention. Happy to take your steer on this.

Reply:
I personally think the benefit to having two characters in this case is larger than the minimal stylistic change of a single character.  Common variables like "err" are longer, and I think `hl` still represents the go-mindset of short variable names.